rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -2412,6 +2435,39 @@ rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * tak
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+rmw_take(const rmw_subscription_t * subscription, void * ros_message, bool * taken)
+{
+  return _take_impl(subscription, ros_message, taken, nullptr);
+}
+
+rmw_ret_t
+rmw_take_with_info(
+  const rmw_subscription_t * subscription,
+  void * ros_message,
+  bool * taken,
+  rmw_message_info_t * message_info)
+{
+  if (!message_info) {
+    RMW_SET_ERROR_MSG("message info is null");
+    return RMW_RET_ERROR;
+  }
+  DDS_InstanceHandle_t sending_publication_handle;
+  auto ret = _take_impl(subscription, ros_message, taken, &sending_publication_handle);
+  if (ret != RMW_RET_OK) {
+    // Error string is already set.
+    return RMW_RET_ERROR;
+  }
+
+  rmw_gid_t * sender_gid = const_cast<rmw_gid_t *>(&message_info->publisher_gid);
```

Comment:
Neither `message_info` nor `publisher_gid` seems to be const. Why is the `const_cast` necessary here?